osbs/build/user_params.py

```diff
@@ -202,22 +162,24 @@ def set_params(self,
         :param reactor_config_map: str, name of the config map containing the reactor environment
         :param worker_deadline: int, worker completion deadline in hours
         """
+        # TODO: see if setting params from build_conf can be separated from setting other params,
+        #   if yes, move most of this method to __init__
```

Comment:
Here's the solutions that I came up with for refactoring `set_params()`:
* make it `__init__()`
  * `+` validation could be done right away during init
  * `-` the `build_conf` argument is not for a BuildParam (other BuildParams are set from build_conf), so it might be confusing
* put most of it into `__init__()`, add a `set_params_from_build_conf()` method
  * `+` separates basic param setting from the additional build_conf logic
  * `-` set_params does some cross-validation between build_conf and normal params (specifically, it checks that `build_from` is set either in build_conf or as a normal param) - this would be lost
  * `-` validation would have to be done by calling `validate()`
* turn it into a classmethod that creates a new instance
  * `+` validation could be done right away in `<name of method>`
  * `-` what should be the name of the method?

Reply:
Tried option 3, see last commit (the validation part is not complete)

Reply:
Looks like there might be a case for not validating during creation: https://github.com/containerbuildsystem/osbs-client/blob/56d648c020948db68c74143086d2cdc06add6aa9/osbs/build/build_requestv2.py#L73-L74

Not sure if there really is a reason for not validating user params, but for now, I'm calling it good enough.